framework/encode/d3d12_capture_manager.cpp

```diff
@@ -1770,21 +1851,156 @@ void D3D12CaptureManager::PreProcess_ID3D12CommandQueue_ExecuteCommandLists(ID3D
         }
     }
 
-    PreQueueSubmit();
+    // Split commandlist is for trim drawcalls. It means that this is a extra ExecuteCommandLists. It shouldn't count
+    // queue_submit_count_.
+    if (!is_split_commandlist)
+    {
+        PreQueueSubmit();
+    }
+}
+
+void D3D12CaptureManager::OverrideID3D12CommandQueue_ExecuteCommandLists(ID3D12CommandQueue_Wrapper* wrapper,
+                                                                         UINT                        num_lists,
+                                                                         ID3D12CommandList* const*   lists)
+{
+    bool is_complete = false;
+    if (IsTrimEnabled() && GetTrimBoundary() == CaptureSettings::TrimBoundary::kDrawCalls)
+    {
+        auto trim_draw_calls = GetTrimDrawCalls();
+        // TODO: When queue_submit_count_ becomes 0-based, remove "-1".
+        if ((common_manager_->GetQueueSubmitCount() - 1) == trim_draw_calls.submit_index)
+        {
+            auto                            target_cmdlist = lists[trim_draw_calls.command_index];
+            ID3D12CommandList_Wrapper*      target_wrapper = nullptr;
+            target_cmdlist->QueryInterface(IID_IUnknown_Wrapper, reinterpret_cast<void**>(&target_wrapper));
+            GFXRECON_ASSERT(target_wrapper);
+
+            auto target_info = target_wrapper->GetObjectInfo();
+            GFXRECON_ASSERT(target_info);
+
+            if (!target_info->is_split_commandlist)
+            {
+                DecrementCallScope();
```

Comment:
In order to write `wrapper->ExecuteCommandLists(cmdlists.size(), cmdlists.data());` that is between `ActivateTrimmingDrawCalls` and `DeactivateTrimmingDrawCalls`.  Since call_scope_ has been 1, it needs to `DecrementCallScope`.

In old version, I ran unwrapped queue and `Encode_ID3D12CommandQueue_ExecuteCommandLists` to write this `ExecuteCommandLists`. This version run wrapped queue to write ExecuteCommandLists.